src/parser/alter.rs

```diff
@@ -36,6 +39,66 @@ impl<'a> Parser<'a> {
         ))
     }
 
+    /// Parse ALTER POLICY statement
+    /// ```sql
+    /// ALTER POLICY policy_name ON table_name [ RENAME TO new_name ]
+    /// or
+    /// ALTER POLICY policy_name ON table_name
+    /// [ TO { role_name | PUBLIC | CURRENT_ROLE | CURRENT_USER | SESSION_USER } [, ...] ]
+    /// [ USING ( using_expression ) ]
+    /// [ WITH CHECK ( check_expression ) ]
+    /// ```
+    ///
+    /// [PostgreSQL](https://www.postgresql.org/docs/current/sql-alterpolicy.html)
+    pub fn parse_alter_policy(&mut self) -> Result<Statement, ParserError> {
+        let name = self.parse_identifier(false)?;
+        self.expect_keyword(Keyword::ON)?;
+        let table_name = self.parse_object_name(false)?;
+
+        let to = if self.parse_keyword(Keyword::TO) {
+            Some(self.parse_comma_separated(|p| p.parse_owner())?)
+        } else {
+            None
+        };
+
+        if self.parse_keyword(Keyword::RENAME) {
```

Comment:
from the doc it looks like the `RENAME` vs others are  are mutually exclusive, so that if we parse a `TO` it probably won't be correct to potentially also parse a rename afterwards as the current code suggests?
e.g `ALTER POLICY a on t TO owner RENAME TO new_name` seems like its accepted by the parser?

Reply:
@iffyio Yes, you're right. I falsely parse TO before REANME, will fix it soon. Thank you for your correction.

Reply:
@iffyio I have fixed this and also added test cases to prevent this happen.